### PR TITLE
Replace std::is_same<..., void> with std::is_void<...>

### PR DESCRIPTION
**Pull request by masterleinad**
Just some cleanup.

### CODE REVIEW

File: core/src/Cuda/Kokkos_Cuda_Parallel_Range.hpp [hunk 1]
```diff
@@ -81,16 +81,14 @@ class ParallelFor<FunctorType, Kokkos::RangePolicy<Traits...>, Kokkos::Cuda> {
   ParallelFor& operator=(const ParallelFor&) = delete;
 
   template <class TagType>
-  inline __device__
-      typename std::enable_if<std::is_same<TagType, void>::value>::type
-      exec_range(const Member i) const {
+  inline __device__ typename std::enable_if<std::is_void<TagType>::value>::type
```
[the remark this answers is not in the document]
Comment (PhilMiller) [reply core/src/Cuda/Kokkos_Cuda_Parallel_Range.hpp:84]: That'll be especially worthwhile to revisit when we adopt C++17, and can say `std::enable_if_t<std::is_void_v<TagType>>`

[the remark this answers is not in the document]
Comment (nliber) [reply core/src/Cuda/Kokkos_Cuda_Parallel_Range.hpp:84]: I concur with both of the above comments.

Comment (nliber) [inline core/src/Cuda/Kokkos_Cuda_Parallel_Range.hpp:84]: I concur with both of the above comments.

**Review by nliber — CHANGES_REQUESTED** [review]

File: core/src/Cuda/Kokkos_Cuda_Parallel_Range.hpp (hunk 1 above)
Comment (PhilMiller) [reply core/src/Cuda/Kokkos_Cuda_Parallel_Range.hpp:84]: To be clear, I was suggesting applying Dong-Hun's further cleanup later, and just going ahead with this now

Comment (nliber) [reply core/src/Cuda/Kokkos_Cuda_Parallel_Range.hpp:84]: I'd rather the `enable_if_t` cleanup be done as part of this (but it isn't a strong preference).  This is already touching 40% of the files where such a change should be made (which can also be an argument for doing that in a separate PR).

Comment (dalg24) [reply core/src/Cuda/Kokkos_Cuda_Parallel_Range.hpp:84]: OK to (optionally) fix the lines that are being touched.  I strongly oppose to a global `enable_if_t` refactoring as part of this PR.
Note that someone (you?) need to take this PR over and apply the changes if you really want the fix on the lines that are being edited.  Daniel is away for a little while.

**Review by cz4rs — APPROVED** [review]
Looks good to me.
Regarding the `enable_if` -> `enable_if_t` change - I can do that in a separate PR to avoid invading this one.

**Review by nliber — APPROVED** [review]

**Comment by masterleinad** [comment]
Retest this please.

**Comment by dalg24** [comment]
It now has conflicts after the merge of #4990

**Comment by masterleinad** [comment]
> It now has conflicts after the merge of #4990

Rebased.

**Pull request merged**